### DISCUSSION TITLE
Move all Invalidation-related events to a distinct queue

### DISCUSSION THREAD

**Discussion by mnlmaier** [statamic/cms#9923] [Q&A]
Hi guys,

Quick question, maybe somebody already came up with something and I'm just missing it. I have a rather complex site, not entry-wise, but lots of maintenance jobs, asset preset generations and imports going on in the background, polluting the default queue occasionally. I don't care how long those tasks take, as they are not super urgent but have to be running periodically.

What I'd like to achieve is some kind of fine-grained control over the default jobs which are dispatched by Statamic. For example, I'd like to have all Invalidation jobs running on a distinct queue, since I'd like to have a seperate Horizon supervisor controlling it, since they are — by far — the most important jobs on the site when people are maintaining the content. Also it just makes sense for the project to start structuring the whole queue mess.

Is there a way to force Statamic to use special queues for its default Jobs? I think it would be great to be able to control this through a config file of some sorts.

Sure, I could probably just use priorit… wait, I can not, no options for that either :D (unless i am missing something)

I there's nothing like that, we'll have to implement it ourselves and patch it in :)

**Comment by edalzell** [comment]
Sounds like a great feature to PR

**Comment by robdekort** [comment]
Oh that sounds pretty great indeed. Also to run mailing + git on a separate queue. I have seen queues crap out when they get to many image processing jobs, and then mails don't get out for months at a time, because the queue is borked.

**Comment by mnlmaier** [comment]
@robdekort @edalzell Glad I'm not the only one! I guess we'll figure something out :)
